### DISCUSSION TITLE
Is there a way to overwrite or set the "build starter user" via the build API?

### DISCUSSION THREAD

**Discussion by nevotheless** [codemagic-ci-cd/codemagic-docs#2197] [Q&A] [answered]
### If you are looking for help, please confirm the following...

- [X] I confirm I have searched the [Docs](https://docs.codemagic.io), [Codemagic Sample Projects](https://github.com/codemagic-ci-cd/codemagic-sample-projects), and GitHub [Discussions](https://github.com/orgs/codemagic-ci-cd/discussions).
- [X] I confirm the software versions on my local computer match the ones in CI environment (e.g. Xcode version, Gradle version). See [here](https://docs.codemagic.io/troubleshooting/common-android-issues/#builds-work-locally-but-fail-on-codemagic)
- [X] I confirm I am able to do a fresh clone of the repository and build the project on my local computer.

### Which mobile framework are you using?

Other - please mention in your question

### Steps to reproduce

1. Create API Key (only personal API keys are possible)
2. Use personal API Key for integrations
3. Now every Build is started by myself which is _technically true_ but not very helpful when i try to build custom integrations for our team

Way to overwrite that is needed. 

### Expected results

Codemagic doesn't seem to provide a way a TEAM Entity can use a shared API Tokens for custom integrations. Even though the build API is vast enough to let it seem that this was a no-brainer to have. Instead team members have to use their personal API Tokens for custom integrations.

I would be fine with it, even though it seems bad practice to now be able to have team wide API Tokens, if i would be able to overwrite or just set the starting user by myself. Either globally for the whole build or just for notifications because they always print my name now which is not helpful for our team.

I would like to have an option to set or overwrite the "build owner/starter".

### Actual results

Since my personal API key is used for our team integrations every build is now "started by me" which is wrong and not helpful. Slack Notifications also say that i started the build.

**Comment by icarusdust** [comment, marked as answer]
Hi @nevotheless, what about creating a general user account for your team that will have its own API Token, then use the given token for all the team builds rather than using each team member's. This way, instead of your name, the general company user's name will be printed out.

**Reply by nevotheless** [reply]
While this is certainly not prefered this should work for now. Haven't thought about this before thanks!
